security/cmd/node_agent_k8s/main.go

```diff
@@ -116,7 +128,7 @@ func newSecretCache(serverOptions sds.Options) (workloadSecretCache, gatewaySecr
 	}
 
 	if serverOptions.EnableIngressGatewaySDS {
-		gSecretFetcher, err := secretfetcher.NewSecretFetcher(true, "", "", false, nil)
+		gSecretFetcher, err := secretfetcher.NewSecretFetcher(true, "", "", serverOptions.IngressSecretNameSpace, false, nil)
```

Comment:
is this right to set client as nil in dev code ? also better to init it in NewSecretFetcher if ```ingressGatewayAgent==true```.